### PR TITLE
[serialport] Allow baudrate change at run time

### PR DESCRIPTION
**Pull request by Orfait**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

Allow baudrate modification during run time of the flow.
https://discourse.nodered.org/t/node-red-node-serialport-change-baudrate-on-the-fly/4313

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red-nodes/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: io/serialport/25-serial.js [hunk 1]
```diff
@@ -249,6 +277,7 @@ module.exports = function(RED) {
                             return payload;
                         },
                         write: function(m,cb) { this.serial.write(m,cb); },
+			update: function(m,cb) { this.serial.update(m,cb); },
```
Comment (dceejay) [inline io/serialport/25-serial.js:280]: alignment

File: io/serialport/25-serial.js [hunk 2]
```diff
@@ -121,6 +135,20 @@ module.exports = function(RED) {
             node.port = serialPool.get(this.serialConfig);
             // Serial Out
             node.on("input",function(msg) {
+                if (msg.hasOwnProperty("baudrate")) {
+					var baud = parseInt(msg.baudrate);
+					if (isNaN(baud)) {
+						var errmsg = err.toString().replace("Serialport","Serialport "+node.port.serial.path);
+						node.error(errmsg,"Cannot parse baudrate property");
```
Comment (dceejay) [inline io/serialport/25-serial.js:142]: as per above errmsg should be the text, 2nd param should just be msg.
It should also be translatable - the text should be in the locale file - eg see line 61 etc.

File: io/serialport/25-serial.js [hunk 3]
```diff
@@ -41,6 +41,20 @@ module.exports = function(RED) {
             node.port = serialPool.get(this.serialConfig);
 
             node.on("input",function(msg) {
+				if (msg.hasOwnProperty("baudrate")) {
+					var baud = parseInt(msg.baudrate);
+					if (isNaN(baud)) {
+						var errmsg = err.toString().replace("Serialport","Serialport "+node.port.serial.path);
+						node.error(errmsg,"Cannot parse baudrate property");
```
Comment (dceejay) [inline io/serialport/25-serial.js:48]: the errmsg should be the text you have, and the second parameter should just be the incoming msg.
They should also be translatable - the text should be in the locale file - eg see line 61 etc.

File: io/serialport/25-serial.js (hunk 2 above)
Comment (Orfait) [reply io/serialport/25-serial.js]: Changed like line 161.

File: io/serialport/25-serial.js (hunk 3 above)
Comment (Orfait) [reply io/serialport/25-serial.js]: Changed to be like line 63

File: io/serialport/25-serial.js [hunk 4]
```diff
@@ -41,6 +41,20 @@ module.exports = function(RED) {
             node.port = serialPool.get(this.serialConfig);
 
             node.on("input",function(msg) {
+				if (msg.hasOwnProperty("baudrate")) {
+					var baud = parseInt(msg.baudrate);
+					if (isNaN(baud)) {
+						var errmsg = err.toString().replace("Serialport","Serialport "+node.port.serial.path);
```
Comment (dceejay) [inline io/serialport/25-serial.js:47]: There is no err at this point - you should just set it to the string you want. But the string needs to be the translated one - so something like `node.error(RED._("serial.errors.badbaudrate"),msg);`
and then create a badbaudrate entry in the US locale file.

Comment (Orfait) [reply io/serialport/25-serial.js]: I did it, but I must admit it is over my actual knowledge.

**Comment by dceejay** [comment]
Thanks

**Comment by Orfait** [comment]
Hi @dceejay 
I forgot to change the version on the node (actually 0.10.3). Is it needed ?
Maybe this is done when making a release ?

**Comment by dceejay** [comment]
no need - yes I will change it... - I also want to update the underlying serialport lib so have included that and am testing before release.

**Pull request merged**